src/pages/resultsView/download/DownloadUtils.ts

```diff
@@ -321,6 +322,79 @@ export function generateOtherMolecularProfileDownloadData(
         : [];
 }
 
+export function generateGenericAssayProfileData(
+    molecularProfileIds: string[],
+    unfilteredCaseAggregatedData: CaseAggregatedData<GenericAssayData>
+): { [key: string]: GenericAssayData[] } {
+    const sampleFilter = (data: GenericAssayData) => {
+        return molecularProfileIds.includes(data.molecularProfileId);
+    };
+
+    // generate GenericAssay profile data by key
+    // key => stableId + uniqueSampleKey
+    const sampleDataByStableId: { [key: string]: GenericAssayData[] } = {};
+
+    _.values(unfilteredCaseAggregatedData.samples).forEach(alterations => {
+        alterations.forEach(alteration => {
+            const key = `${alteration.stableId}_${alteration.uniqueSampleKey}`;
+            sampleDataByStableId[key] = sampleDataByStableId[key] || [];
+
+            // alteration is filtered out if filter function returns false
+            if (sampleFilter(alteration)) {
+                sampleDataByStableId[key].push(alteration);
+            }
+        });
+    });
+    return sampleDataByStableId;
+}
+
+export function generateGenericAssayProfileDownloadData(
+    sampleGenericAssayDataByStableId: { [key: string]: GenericAssayData[] },
+    samples: Sample[] = [],
+    stableIds: string[] = []
+): string[][] {
+    if (_.isEmpty(sampleGenericAssayDataByStableId)) {
+        return [];
+    } else {
+        // we need the sample index for better performance
+        const sampleIndex = _.keyBy(samples, 'uniqueSampleKey');
+        const sampleKeys = samples.map(sample => sample.uniqueSampleKey);
+
+        // generate row data (keyed by uniqueSampleKey)
+        const rows = generateGenericAssayDownloadFileRows(
+            sampleGenericAssayDataByStableId,
+            stableIds,
+            sampleIndex,
+            sampleKeys
+        );
+
+        const downloadData: string[][] = [];
+
+        // add headers
+        downloadData.push(['STUDY_ID', 'SAMPLE_ID'].concat(stableIds));
+
+        // convert row data into a 2D array of strings
+        sampleKeys.forEach(sampleKey => {
+            const rowData = rows[sampleKey];
+            const row: string[] = [];
+
+            row.push(rowData.studyId);
+            row.push(rowData.sampleId);
+
+            stableIds.forEach(stableId => {
+                // format: space delimited join
+                const formattedValue =
+                    rowData.alterationData[stableId].join(' ') || 'NA';
```

Comment:
i think you mean to catch the condition where stableId doesn't exist in alterationData, and instead return NA.  but this will error because of call to join on undefefined.

Reply:
The `alterationData[stableId]` will never be undefined since we will put an empty list there, please see logic at here: https://github.com/cBioPortal/cbioportal-frontend/pull/3284/files#diff-fc7a2c223844f7cba7399ed39f7813adR482. I can put a comment here if that will make this more clear.